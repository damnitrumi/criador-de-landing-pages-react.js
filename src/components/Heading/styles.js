import styled from "styled-components";
import { css } from "styled-components";

export const Title = styled.h1`
  ${({ theme, light }) => css`
    color: ${light ? theme.colors.white : theme.colors.primaryColor};
  `}
`;
