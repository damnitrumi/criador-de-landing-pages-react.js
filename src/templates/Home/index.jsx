import { Heading } from "../../components/Heading";
import * as Styled from "./styled";

function Home() {
  return (
    <div className="App">
      <Styled.Wrapper>
        <Heading>O texto que eu quero</Heading>
      </Styled.Wrapper>
    </div>
  );
}

export default Home;
