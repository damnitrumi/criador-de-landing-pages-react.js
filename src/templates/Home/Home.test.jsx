import Home from "./index";
import { renderTheme } from "../../styles/render-theme";
import { screen } from "@testing-library/react";
import { theme } from "../../styles/theme";

it("should render with background mainBg", () => {
  renderTheme(<Home />);

  const headingContainer = screen.getByRole("heading", {
    name: "Hello!",
  }).parentElement;

  expect(headingContainer).toHaveStyle({
    background: theme.colors.mainBg,
  });

  expect(headingContainer).toHaveStyleRule("background", "red");

  expect(headingContainer).toMatchSnapshot();
});
