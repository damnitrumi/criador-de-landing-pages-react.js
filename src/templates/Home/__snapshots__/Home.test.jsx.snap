// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with background mainBg 1`] = `
.c0 {
  background: red;
}

<div
  class="c0"
>
  <h1>
    Hello!
  </h1>
</div>
`;
